src/bots/spotFiller.ts

```diff
@@ -244,6 +247,9 @@ export class SpotFillerBot implements Bot {
 
 	private periodicTaskMutex = new Mutex();
 
+	protected hasEnoughSolToFill: boolean = true;
+	protected hasEnoughSolToFillMutex = new Mutex();
```

Comment:
i dont think this mutex is necessary. Especially since it's getting acquired on every fill, the overhead is probably not worth the synchronization benefits